### ISSUE TITLE
MimeMagic.by_magic returns nil for Javascript file

### ISSUE THREAD

**Issue by etagwerker** [minad/mimemagic#30]
Hi,

I was trying to detect the type for a simple js file and this is what I'm getting:

```
2.1.2 :028 > @filename = "/home/deployer/ajax.js"
 => "/home/deployer/ajax.js"
2.1.2 :029 > MimeMagic.by_magic(File.open(@filename)).try(:type)
 => nil
```

Here is more information about the file: 

```
[deployer@production ~]$ ls -la ajax.js
-rw-rw-r--. 1 deployer deployer 316 Dec 22 14:35 ajax.js
[deployer@production ~]$ pwd
/home/deployer
[deployer@production ~]$ cat ajax.js
jQuery(document).ready(function(){

  $("#cart-form form").bind('ajax:success', function(data, response, status, xhr) {
    console.log(xhr.responseText);
    console.log("Great Success!");
  });

  $("#cart-form form").bind('ajax:error', function(data, response, status) {
    console.log("Problem! ");
  });

});
```

I expected it to return something like "application/javascript", "application/x-javascript" or "text/plain" 

This is what I get if I use `file`:

```
[deployer@production ~]$ file -b --mime-type ajax.js
text/plain
```

It seems to be working fine for other extensions:

```
2.1.2 :024 > @filename = "/home/deployer/prune-var-backups.sh"
 => "/home/deployer/prune-var-backups.sh"
2.1.2 :025 > MimeMagic.by_magic(File.open(@filename)).try(:type)
 => "application/x-shellscript"
2.1.2 :026 > @filename = "/home/deployer/cerdo.jpg"
 => "/home/deployer/cerdo.jpg"
2.1.2 :027 > MimeMagic.by_magic(File.open(@filename)).try(:type)
 => "image/jpeg"
```

Do you know what could be wrong?

**Comment by scpike** [comment]
@etagwerker not to speak for @minad, but here's my POV: 

This library's initial intent was to use the free desktop.org's "magic" list to do content type detection. They only do JS detection by extension (http://cgit.freedesktop.org/xdg/shared-mime-info/tree/freedesktop.org.xml.in?id=Release-1-5). 

For xlsx, I found extra magic by looking through binary dumps of a bunch of excel files. That magic wasn't acceptable to the free desktop project (I have to look too far into the file, and their project has strict performance constraints) so I proposed adding it as an overlay here.  https://github.com/minad/mimemagic/blob/master/lib/mimemagic/overlay.rb#L3

We could add a new overlay for JS magic, but someone needs to find it.
